source/features/clone-branch.tsx

```diff
@@ -0,0 +1,68 @@
+import gitBranch from 'octicon/git-branch.svg';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getCleanPathname} from '../libs/utils';
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void|false> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = (currentTarget.closest('[data-branch-name]') as HTMLAnchorElement);
+	const newBranchName = prompt('Enter the new branch name');
+	if (!newBranchName) {
+		return false;
+	}
+
+	const spinner = select('.js-loading-spinner', branchElement)!;
+	spinner.hidden = false;
+	currentTarget.hidden = true;
+
+	const [user, repo] = getCleanPathname().split('/');
+	const getBranchInfo = await api.v3(`repos/${user}/${repo}/git/refs/heads/${branchElement.dataset.branchName!}`);
+	try {
+		await api.v3(`repos/${user}/${repo}/git/refs`, {
+			method: 'POST',
+			body: {
+				sha: String(getBranchInfo.object.sha),
+				ref: 'refs/heads/' + newBranchName
+			}
+		});
+		branchElement.after(branchElement.cloneNode(true));
+	} catch (error) {
+		const errorMessage = JSON.parse(error.message.slice(Math.max(0, error.message.indexOf('{')))).message;
```

Comment:
What.

Reply:
When you throw the error you add a whole bunch of text

Reply:
Right, why are you removing it? What's the point to have it there at all?

In `revert-files` I used something like "error found, check the console"

Reply:
Ok